gendarme-ignore.txt

```diff
@@ -16,6 +23,10 @@ R: Gendarme.Rules.BadPractice.PreferSafeHandleRule
 T: Java.Interop.JniEnvironmentInfo
 T: Java.Interop.JniRuntime
 
+R: Gendarme.Rules.Correctness.AvoidFloatingPointEqualityRule
+# This is in from generated code. We could plausibly fix the generated code, but I'm not at all sure what to change it *to*.
+M: System.Int32 Java.Interop.JavaSingleArray::IndexOf(System.Single)
+M: System.Int32 Java.Interop.JavaDoubleArray::IndexOf(System.Double)
 
```

Comment:
It should be used with given values.

For example, you'd do:

    var array = new JavaSingleArray (new[]{1.0f, 2.0f, 3.0f});
    Assert.AreEqual (0, array.IndexOf (1.0f));

More elaborately, Java code would have `float[] a = new float[]{...}` and send `a` via JNI into managed code. Managed code will receive it as a `jobject` (`IntPtr`), and "wrap" via the `JavaSingleArray (ref JniObjectReference, JniObjectReferenceOptions options)` constructor, at which point I'd *also* expect `.IndexOf()` to work with the "same" value as was provided in Java.

This is assuming that Android/Dalvik and Mono use the same floating-point representation, which *should* be the case.